File: 07. Servicios y procesos/WorkSpace/EjerciciosCasa/src/ejercicioElectrodomesticos/Electrodomestico.java
package ejercicioElectrodomesticos;

public class Electrodomestico {
	//Atributos
	protected float precio_base; //€
	protected String color;
	protected char consumo;
	protected float peso; //Kilo
	
	//Atributos por defecto
	private final float PRECIO_DEFECTO = 100; 
	private final String COLOR_DEFECTO = "blanco";
	private final char CONSUMO_DEFECTO = 'F';
	private final float PESO_DEFECTO = 5; 
	
	private final String[] COLORES_DISPONIBLES = {"blanco", "negro", "rojo", "azul", "gris"};
	
	//Constructores
	public Electrodomestico() {
		precio_base = PRECIO_DEFECTO;
		color= COLOR_DEFECTO;
		consumo = CONSUMO_DEFECTO;
		peso = PESO_DEFECTO;
	}
	
	public Electrodomestico(float precio_base, float peso) {
		super();
		this.precio_base = precio_base;
		this.peso = peso;
		color= COLOR_DEFECTO;
		consumo = CONSUMO_DEFECTO;
	}
	
	public Electrodomestico(float precio_base, String color, char consumo, float peso) {
		super();
		this.precio_base = precio_base;
		this.color = comprobarColor(COLORES_DISPONIBLES, color);
		this.consumo = comprobarConsumoEnergetico(consumo);
		this.peso = peso;
	}
	
	//Métodos get
	public float getPrecio_base() {
		return precio_base;
	}
	public String getColor() {
		return color;
	}
	public char getConsumo() {
		return consumo;
	}
	public float getPeso() {
		return peso;
	}
	
	//Otros métodos
	private String comprobarColor(String[] disponibles, String color) {
		for (String colorValido:disponibles) {
			if(color.toLowerCase() == colorValido) {
				return color.toLowerCase();
			}
		}
		return COLOR_DEFECTO;
	}
	
	private char comprobarConsumoEnergetico(char letra){
		letra = Character.toUpperCase(letra);
		if(letra >= 'A'  && letra <= 'F') {
			return letra;
		}
		return CONSUMO_DEFECTO;
	}
	
	public double precioFinal() {
		double preFinal = precio_base;
		switch(consumo) {
		case 'A':
			preFinal += 100;
			break;
		case 'B':
			preFinal += 80;
			break;
		case 'C':
			preFinal += 60;
			break;
		case 'D':
			preFinal += 50;
			break;
		case 'E':
			preFinal += 30;
			break;
		case 'F':
			preFinal += 10;
			break;
		}
		
		if(peso > 0 && peso <20) {
			preFinal +=10;
		}else if(peso >20 && peso <50) {
			preFinal +=50;
		}else if(peso >50 && peso <80) {
			preFinal +=80;
		}else if(peso > 80) {
			preFinal +=100;
		}
	return preFinal;
	}

	@Override
	public String toString() {
		return "Electrodomestico [precio_base=" + precio_base + "€, color=" + color + ", consumo=" + consumo + ", peso="
				+ peso + "kg]";
	}

}

File: 06. Python/POO_001/POO/POO_EJER_3/Electrodomestico.py
from enum import Enum

class ConsumoEnergetico(Enum):
    A = 100
    B = 80
    C = 60
    D = 50
    E = 30
    F = 10

class Electrodomestico:

    COLOR_DEFECTO = "blanco"
    CONSUMO_DEFECTO = ConsumoEnergetico.F
    PRECIO_BASE_DEFECTO = 100
    PESO_DEFECTO = 5

    COLORES_DISPONIBLES = ["blanco", "negro", "rojo", "azul", "gris"]

    def __init__(self, precioBase=PRECIO_BASE_DEFECTO, peso=PESO_DEFECTO,
                 color=COLOR_DEFECTO, consumoEnergetico=CONSUMO_DEFECTO):

        self.precioBase = precioBase
        self.peso = peso
        self.color = self.comprobarColor(color)
        self.consumoEnergetico = self.comprobarConsumoEnergetico(consumoEnergetico)

    # getters
    @property
    def precio(self):
        return self.precioBase

    @precio.setter
    def precio(self, precio):
        self.precioBase = precio

    @property
    def color(self):
        return self.color

    @color.setter
    def color(self, color):
        self.color = color

    @property
    def consumoEnergetico(self):
        return self.consumoEnergetico
    @consumoEnergetico.setter
    def consumoEnergetico(self, consumoEnergetico):
        self.consumoEnergetico = consumoEnergetico

    @property
    def peso(self):
        return self.peso

    @peso.setter
    def peso(self, peso):
        self.peso = peso

    def comprobarConsumoEnergetico(self, consumo):
        if isinstance(consumo, ConsumoEnergetico):
            return consumo
        return self.CONSUMO_DEFECTO

    def comprobarColor(self, color):
        color = color.lower()
        if color in self.COLORES_DISPONIBLES:
            return color
        return self.COLOR_DEFECTO

    def precioFinal(self):

        precio = self.precioBase

        # aumento por consumo (usamos el enum)
        precio += self.consumoEnergetico.value

        # aumento por peso
        match True:
            case _ if self.peso >= 80:
                precio += 100
            case _ if self.peso >= 50:
                precio += 80
            case _ if self.peso >= 20:
                precio += 50
            case _:
                precio += 10

        return precio

if __name__ == "__main__":
    ...